src/main/resources/api/swagger.yaml

```diff
@@ -1,74 +1,80 @@
 swagger: '2.0'
 info:
   title: Pazuzu registry
-  description: Generating Dockerfiles by sets of required functions
-  version: "0.0.4"
+  description: Generating Dockerfiles by sets of required functions.
+  version: "0.0.5"
 host: pazuzu.zalando.net
 schemes:
   - http
 produces:
   - application/json
 paths:
-  /api/resolved-features:
-    get:
-      summary: Resolve all features by name
-      description: |
-        Returns all the features with all its dependencies in a single array.
-      tags:
-        - Features
-      parameters:
-        - name: names
-          in: query
-          description: feature names.
-          required: true
-          type: array
-          items:
-            type: string
-      responses:
-        200:
-          description: An array of features
-          schema:
-            type: array
-            items:
-              $ref: '#/definitions/Feature'
-        default:
-          description: Unexpected error
-          schema:
-            $ref: '#/definitions/Error'
   /api/features:
     get:
-      summary: Get all features
+      summary: Get features
       description: |
-        Returns all the features without its dependencies.
+        Returns the features.
       tags:
         - Features
       parameters:
         - name: names
           in: query
-          description: value, that must present in feature name.
+          description: Exact names of the features the call should return.
           required: false
           type: array
+          maxItems: 128
           items:
             type: string
+        - name: resolve
```

Comment:
It's ok to model it this way.

Another more rest-ish way would be to treat dependencies of each feature as just another attribute of a feature which can be embedded. See embedding in our guidelines for it, http://zalando.github.io/restful-api-guidelines/performance/Performance.html

Dependencies of a feature could be a sub-resource: /feature/{feature_id}/dependencies And you could resolve them by embedding: /feature/{feature_id}?embed=(dependencies)

This way the resolving would not be model as a process which is triggered with a boolean flag, it would be part of your resources.